lib/Common/Memory/LargeHeapBucket.cpp

```diff
@@ -126,17 +124,22 @@ LargeHeapBucket::PageHeapAlloc(Recycler * recycler, size_t size, ObjectInfoBits
         return nullptr;
     }
 
-    size_t actualPageCount = pageCount + 1; // for page heap
+    if(size<sizeof(void*))
```

Comment:
if alloc size is less than pointer size(eg: ask for 3 bytes), it must be a leaf object. we have runtime code and trident code currently not passing leaf attribute for such allocation. 
in page heap mode, I force it to be a leaf, so don't mark it.

I think we should assert or force it to be leaf in non-page heap mode as well. this reduces false positive in x86


---
In reply to: [60990204](https://github.com/Microsoft/ChakraCore/pull/711#discussion_r60990204) [](ancestors = 60990204)